project/Component/NefModels/ProgressReport.swift

```diff
@@ -0,0 +1,52 @@
+import BowEffects
+
+public protocol ProgressReport {
+    func notify<E: Error, A: CustomProgressDescription>(_ event: ProgressEvent<A>) -> IO<E, Void>
+}
+
+public extension ProgressReport {
+    func oneShot<E: Error, A: CustomProgressDescription>(_ step: A) -> IO<E, Void> {
+        self.notify(
+            ProgressEvent(step: step,
+                          status: .oneShot))
+    }
+    
+    func inProgress<E: Error, A: CustomProgressDescription>(_ step: A) -> IO<E, Void> {
+        self.notify(
+            ProgressEvent(step: step,
+                          status: .inProgress))
+    }
+    
+    func succeeded<E: Error, A: CustomProgressDescription>(_ step: A, info: String = "") -> IO<E, Void> {
+        self.notify(
+            ProgressEvent(step: step,
+                          status: .successful(info: info)))
+    }
+    
+    func failed<E: Error, A: CustomProgressDescription>(_ step: A, _ error: E, info: String = "") -> IO<E, Void> {
+        self.notify(
+            ProgressEvent(step: step,
+                          status: .failed(error, info: info)))
+    }
+}
+
+public enum ProgressEventStatus {
+    case oneShot
+    case inProgress
+    case successful(info: String)
```

Comment:
You do not need `info` in success state. It will come from `inProgress` or directly in `oneShot`
```suggestion
    case successful
```